File: TPs/TP1/Cesar/Cesar.java
/**
 * @author Natanael Barbosa
 */
import java.util.Scanner;

public class Cesar {
	/**
	 * Criptografa uma string usando a cifra de cesar (na tabela ASCII) adicionando, para cada caractere, um valor inteiro cod.
	 */
	public static String cifrar(String str, int cod) {	
		String new_str = "";

		for (int i = 0; i < str.length(); i++) {
			int ch = (int) str.charAt(i); 
			char new_ch = (char)(ch+cod);
			
			new_str += new_ch;		
		}
		
		return new_str;
	}
	
	public static boolean compare(String str1, String str2) {
		boolean res = false;

		if (str1.length() == str2.length() && (str1 != null && str2 != null)) {
			res = true;
			for (int i = 0; i < str1.length(); i++) {
				res = res && str1.charAt(i) == str2.charAt(i);
			}
		}

		return res;
	}

	public static void main(String[] args) {
		Scanner inp = new Scanner(System.in);
		String new_str = "";

		String str = inp.nextLine();
		while (!compare(str, "FIM")) {
			new_str = cifrar(str, 3);	
			System.out.println(new_str);
			str = inp.nextLine();
		}
		inp.close();
	}
}

File: TPs/TP1/Cesar_Recursivo/cesar.c
/**
 * @author Natanael Barbosa
 */
#include <stdio.h>

int get_len(char * str) {
	int res = 0;

	while (str[res] != '\0') res++;

	return res;
}

int compare (char * str1, char * str2) {
	int res = 0;

	if (str1 != NULL && str2 != NULL) {
		if (get_len(str1) != get_len(str2)) res++;
		else {
			for (int i = 0; i < get_len(str1); i++) {
				if (str1[i] != str2[i]) { res++; i = get_len(str1); }
			}
		}
	}
	return res;
}

/**
 * Recebe uma string para criptografá-la somando o caractere com o valor cod. o inteiro i define o indice do caractere da string.
 */
void cesar (char * str, int i, int cod) {
	if (str[i] != '\0') {
		str[i] = str[i]+cod;
		cesar(str, i+1, 3);
	}
}

int main () {
	char str [1000];
	
	scanf("%[^\n]", str); getchar();
	while (compare(str, "FIM") != 0) {
		cesar(str, 0, 3);
		printf("%s\n", str);
		scanf("%[^\n]", str); getchar();
	}

	return 0;
}

